haskell/src/Monocle/Api/Config.hs

```diff
@@ -179,3 +179,31 @@ getTenantGroups Index {..} = Map.toList $ foldr go mempty (fromMaybe [] idents)
     addUser name groupName acc =
       let users = fromMaybe [] (Map.lookup groupName acc)
        in Map.insert groupName (users <> [name]) acc
+
+-- | Get the Ident name for a Given alias
+--
+-- >>> :{
+--  let
+--    index = emptyTenant "test" [createIdent "alice" ["opendev.org/Alice Doe/12345", "github.com/alice89"] []]
+--  in getIdentByAlias index "github.com/alice89"
+-- :}
+-- Just "alice"
+
+-- >>> :{
+--  let
+--    index = emptyTenant "test" [createIdent "alice" ["opendev.org/Alice Doe/12345", "github.com/alice89"] []]
+--  in getIdentByAlias index "github.com/ghost"
+-- :}
+-- Nothing
+getIdentByAlias :: Index -> Text -> Maybe Text
+getIdentByAlias Index {..} alias = getIdentByAliasFromIdents' alias =<< idents
+  where
+    getIdentByAliasFromIdents' = flip getIdentByAliasFromIdents
+
+getIdentByAliasFromIdents :: [Ident] -> Text -> Maybe Text
+getIdentByAliasFromIdents idents' alias = case isMatched <$> idents' of
+  [] -> Nothing
+  x : _ -> x
+  where
+    isMatched :: Ident -> Maybe Text
+    isMatched Ident {..} = if alias `elem` aliases then Just ident else Nothing
```

Comment:
nit: the case analysis looks like the `head :: [a] -> Maybe a` from relude, so that function could be: `join $ head $ map isMatched idents'`, and if we re-arrange the parameters then we don't have to bind idents' using point free like this:

```suggestion
getIdentByAliasFromIdents :: Text -> [Ident] -> Maybe Text
getIdentByAliasFromIdents alias = join . head . map isMatched
  where
    isMatched :: Ident -> Maybe Text
    isMatched Ident {..} = if alias `elem` aliases then Just ident else Nothing
```

Reply:
Here is why: https://github.com/change-metrics/lentille/pull/33/commits/3b0b7a483d35f086b6e5804e97f2bd7912975f15#diff-71dc47f1a59c491b9a150e9ad3ebdaa1a6649b745d5c27903f0fea1042da7ebeR82